### PR TITLE
Remove method

### PR DESCRIPTION
**Pull request by DeepSnowNeeL**
as discussed in #463
We will need to do some tests, if someone want's to help :D :D

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/ast/Node.java [hunk 1]
```diff
@@ -369,4 +372,37 @@ public void tryAddImportToParentCompilationUnit(Class<?> clazz) {
...
+        Class<?> parentClass = parentNode.getClass();
+        while (parentClass != Object.class) {
+            for (Field f : parentClass.getDeclaredFields()) {
+                f.setAccessible(true);
+                try {
+                    Object object = f.get(parentNode);
+                    if (object == null)
+                        continue;
+                    if (List.class.isAssignableFrom(object.getClass())) {
+                        List<?> l = (List<?>) object;
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/Node.java]: Can you use Collection here?

**Comment by matozoid** [comment]
Here, I'll help with the tests. We need...
- a test when parent is null.
- a test when the node is in a collection.
- a test when the node is a normal child.

:-P

**Comment by DeepSnowNeeL** [comment]
Haha :P I mean write thoses ! To be correct we would need to write a test
per node type inside every possible parent node type
#imgoingtodie #nope #plzhelp #hashtag :P

But allright for 3basic tests, I guess we will have issues with "remove
does not work" > "Check the return value and provide a minimal test case
plz" > "oh allright this case wasn't handled !"

It would be great to check at least all the fields that we have in nodes to
confirm that it only contains Collection subtypes or
reference-equals-ok-to-check types

**Comment by matozoid** [comment]
My energy is going towards 3.0, so keep looking for someone to do the boring part of your work ;)

**Comment by matozoid** [comment]
Here, maybe someone from hacktoberfest will help you out :)

**Comment by matozoid** [comment]
@DeepSnowNeeL - this PR is about to miss its second release chance!

**Comment by DeepSnowNeeL** [comment]
What does it need to get into the release ? Just a few tests ? I might be
able to write some basic tests sunday :)

Le sam. 8 oct. 2016 à 19:52, Danny van Bruggen notifications@github.com a
écrit :

**Comment by matozoid** [comment]
Yep, cool!

File: javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithStatements.java [hunk 2]
```diff
@@ -43,4 +43,25 @@ public default T addStatement(int index, final Expression expr) {
         return addStatement(index, stmt);
     }
 
+    public default <A extends Statement> A addAndGetStatement(A statement) {
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithStatements.java]: Intellij tells me that "public" is not necessary here.

Comment (DeepSnowNeeL) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithStatements.java]: As in every method in an interface :p I don't know what's the right/standard way, i'm going to google a bit

Comment (DeepSnowNeeL) [reply javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithStatements.java]: Allright http://stackoverflow.com/questions/161633/ let's remove them :D

**Comment by matozoid** [comment]
I'm going to assume this is done and merge it :)

**Pull request merged**